CONTRIBUTING.md

```diff
@@ -45,22 +45,119 @@ To add a new plugin you need to do the following.
-    - Otherwise, you should add yourself to our [`maintainers.nix`](lib/maintainers.nix) list.
-    - See the documentation at the top of the [`nixpkgs` maintainers list][nixpkgs-maintainers] for more detail.
-  - `settingsOptions`: All or some (only the most important ones) option declarations for this plugin settings.\
-    See below for more information
-  - `settingsExample`: An example of what could the `settings` attrs look like.
-
-[nixpkgs-maintainers]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix
+3. Add the necessary parameters for the [`mkNeovimPlugin`](#mkneovimplugin)/[`mkVimPlugin`](#mkvimplugin)
+
+#### `mkNeovimPlugin`
+
+The `mkNeovimPlugin` function provide a standardize way to create a `Neovim` plugin.
+
+| Parameter                  | Description                                                                                                                                                                                                 | Required | Default Value |
+|----------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|---------------|
+| **name**                   | The name of the plugin.                                                                                                                                                                                      | Yes      | N/A           |
+| **maintainers**            | Maintainers for the plugin.                                                                                                                                                                                  | Yes      | N/A           |
+| **url**                    | The URL of the plugin's repository.                                                                                                                                                                          | Yes      | If not set, `package`'s' `homepage` attribute |
+| **package**                | The nixpkgs package attr for this plugin. Can be a string, a list of strings, a module option, or any derivation. For example, "foo-bar-nvim" for `pkgs.vimPlugins.foo-bar-nvim`, or `[ "hello" "world" ]` will be referenced as `pkgs.hello.world`. | No       | `name` parameter |
+| **imports**                | Additional Nix modules to import.                                                                                                                                                                            | No       | `[]`            |
+| **description**            | A brief description of the plugin.                                                                                                                                                                           | No       | `null`          |
```

Comment:
```suggestion
| **description**            | A brief description of the plugin. Can also be used for non-normative documentation, warnings, tips and tricks.                                                                                                                                                                           | No       | `null`          |
```

Reply:
Looking around online, it seems GFM may not support wrapping/unwrapping table lines and if we wanted this we'd have to resort to HTML tables...

E.g. no mention in the [GFM spec](https://github.github.com/gfm/#tables-extension-)
And no usuful answers on this [Stack Overflow thread](https://stackoverflow.com/questions/15955160/markdown-table-with-long-lines)